### PR TITLE
[Build] Set rpath to load C shared objects on linux

### PR DESCRIPTION
**Pull request by aciidb0mb3r**
Bug: https://bugs.swift.org/browse/SR-1936

### CODE REVIEW

**Comment by aciidb0mb3r** [comment]
@swift-ci please test

**Comment by aciidb0mb3r** [comment]
@swift-ci please test

File: Tests/Xcodeproj/FunctionalTests.swift [hunk 1]
```diff
@@ -133,7 +133,7 @@ func write(path: String, write: (OutputByteStream) -> Void) throws {
 func XCTAssertXcodeBuild(project: String, file: StaticString = #file, line: UInt = #line) {
     var out = ""
     do {
-        try popen(["env", "-u", "TOOLCHAINS", "xcodebuild", "-project", project], redirectStandardError: true) {
+        try popen(["env", "-u", "TOOLCHAINS", "xcodebuild", "-project", project, "-alltargets"], redirectStandardError: true) {
```
Comment (ddunbar) [inline Tests/Xcodeproj/FunctionalTests.swift]: Just curious, is this related to this PR?

Comment (aciidb0mb3r) [inline Tests/Xcodeproj/FunctionalTests.swift]: Not directly but the fixture I modified (added another target) started failing a test because all targets were not being built by xcodebuild.
